codalab/model/bundle_model.py

```diff
@@ -420,8 +420,16 @@ def make_condition(key, field, value):
                 return field == value
             return None
 
+        def add_join(join_table, join_condition):
+            join_tables.append(join_table)
+            join_conditions.append(join_condition)
+
         shortcuts = {'type': 'bundle_type', 'size': 'data_size', 'worksheet': 'host_worksheet'}
 
+        join_tables = []
```

Comment:
Could you add type annotations?

Reply:
Added for the two arrays I'm using (joins and where_clause_conjuncts). 

When I try to add type annotations for some of the other variables (like limit), I get a strange error from mypy:
Deferral trace:
    codalab.model.bundle_model:409
    codalab.model.bundle_model:412
    codalab.model.bundle_model:414
    codalab.model.bundle_model:417
    codalab.model.bundle_model:419
    codalab.model.bundle_model:433
    codalab.model.bundle_model:438
    codalab.model.bundle_model:438
    codalab.model.bundle_model:438
codalab/model/bundle_model.py: error: INTERNAL ERROR: maximum semantic analysis iteration count reached

So, I've left those un-typed for now, but I'll look into this.